main.py

```diff
@@ -1,39 +1,104 @@
 import argparse
 import json
 import logging
+import fnmatch
 
 from lm_eval import tasks, evaluator
 
 logging.getLogger("openai").setLevel(logging.WARNING)
 
+class MultiChoice:
+    def __init__(self, choices):
+        self.choices = choices
+
+    # Simple wildcard support (linux filename patterns)
+    def __contains__(self, values):
+        for value in values.split(","):
+            if len(fnmatch.filter(self.choices, value)) == 0:
+                return False
+
+        return True
+
+    def __iter__(self):
+        for choice in self.choices:
+            yield choice
+
+# Get task base classes for filtering
+task_types = list(set([task.__bases__[0].__name__ for task in tasks.TASK_REGISTRY.values()]))
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument('--model', required=True)
     parser.add_argument('--model_args', default="")
-    parser.add_argument('--tasks', default="all_tasks")
+    parser.add_argument('--tasks', default=None, choices=MultiChoice(tasks.ALL_TASKS))
+    parser.add_argument('--task_type', default=None, choices=MultiChoice(task_types))    
```

Comment:
why these changes?

Reply:
is the MultiChoice just for wildcards? still unclear why we need task_type

Reply:
my understanding for task_type rn is that this is for doing tasks based on what they inherit from, but this doesnt really seem like the best way to handle metatasks imo (i.e nobody really evals on "all HFTasks" or "all MultipleChoiceTasks")

Reply:
Yes, MultiChoice is used for wildcards. The task_type was because I wanted to be able to group tasks, however on reflection using class hierarchy to do this wasn't a great decision, it's probably better to use a config file or something, what would you use?

Reply:
I think we should remove task_type and task_type for now while we figure out how to do meta tasks properly. I'm thinking something that behaves like a regular Task object